bson-adt-core/src/main/scala/adt/bson/BsonValue.scala

```diff
@@ -96,13 +96,7 @@ object BsonPrimitive {
   }
 }
 
-/**
- * TODO: Expand this out in the next major version.
- *
- * Soon all numeric [[BsonValue]]s extend from this trait.
- */
-sealed trait BsonNumber extends BsonValue
```

Comment:
This change was premature and breaks backwards compatibility. Some day, I will redefine a BsonNumber as a base class of all the other number types.